Material.Blazor/Components/Dialog/MBDialog.razor.cs

```diff
@@ -182,73 +153,25 @@ public async Task<string> ShowAsync()
         /// removing the Blazor markup.
         /// </summary>
         /// <returns></returns>
-        public async Task HideAsync()
+        public Task HideAsync()
         {
-            if (IsOpen)
-            {
-                await JsRuntime.InvokeVoidAsync("MaterialBlazor.MBDialog.hide", DialogElem);
-                IsOpen = false;
-                dialogHasInstantiated = false;
-                StateHasChanged();
-            }
-            else
-            {
-                throw new InvalidOperationException("Cannot hide MBDialog that is not already open");
-            }
+            return JsRuntime.InvokeVoidAsync("MaterialBlazor.MBDialog.hide", DialogElem);
         }
 
+        private readonly TaskCompletionSource OpenedSource = new();
+        internal Task Opened => OpenedSource.Task;
+
 
         [JSInvokable]
         public void NotifyOpened()
         {
-            AfterDialogInitialization = true;
-            StateHasChanged();
+            OpenedSource.TrySetResult();
```

Comment:
This essentially "releases" a semaphore (without being a semaphore). Components within the dialog will perform no JS interop until this is set.